### PR TITLE
Fix pyenv FAQ env var commands

### LINKED ISSUE #141: pyenv 'command not found' with system python set
**Describe the bug**
Encountering error running `py` due to pyenv installed and set to `system`

```
$ py
pyenv: python3.9: command not found

The `python3.9' command exists in these Python versions:
  3.9.2

Note: See 'pyenv help global' for tips on allowing both
      python2 and python3 to be found.
```

**To Reproduce**
1. Install `pyenv`
2. pyenv install 3.9.2
3. cargo install python-launcher
4. py

Tried export `PY_PYTHON` as per FAQ but version file does not exist if global not set:
```
head: cannot open '~/.pyenv/version' for reading: No such file or directory
```
So tried creating file with `pyenv global` but that also failed on the special string `system`: 
```
Error parsing 'system' as an integer: invalid digit found in string
```

**Expected behaviour**
Opens system python 3.8 prompt

**Screenshots**

```
$ pyenv versions
* system (set by ~/.pyenv/version)
  3.9.2
```

```
$ py --list
 3.9 │ ~/.pyenv/shims/python3.9 
 3.8 │ /usr/bin/python3.8                 
 3.6 │ /usr/bin/python3.6                 
 2.7 │ /usr/bin/python2.7 
```

**System Details (please complete the following information):**
 - OS: Ubuntu 20.04
 - Shell: bash
 - Launcher Version: 1.0.0

### PR DESCRIPTION
**Pull request by cas--**
When using pyenv and global not set (defaulting to system) or global set to `system` the Launcher was unable to determine correct version to use.

Fixed by using `pyenv exec` to correctly determine current Python version instead of parsing pyenv config files.

Fixes #141

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -221,20 +221,21 @@ but [U+2502/"Box Drawings Light Vertical"](https://www.compart.com/en/unicode/U+
...
+versions, you'll want to set the version the Launcher to the pyenv 
+[global version](https://github.com/pyenv/pyenv/blob/master/COMMANDS.md#pyenv-global).
 
-You can add this line to your `.zshrc` or `.bashrc` file:
+Add these lines to your `.zshrc` or `.bashrc` file:
 
 ```sh
-export PY_PYTHON=$(head -n 1 $(pyenv root)/version | cut -d "." -f 1,2)
+export PY_PYTHON=$(pyenv exec python -c "import sys; print('.'.join(map(str, sys.version_info[0:2])))")
+export PY_PYTHON3=$(pyenv exec python3 -c "import sys; print(*sys.version_info[0:2], sep='.')")
```
Comment (brettcannon) [inline README.md:231]:
```suggestion
export PY_PYTHON=$(pyenv exec python3 -c "import sys; print(*sys.version_info[0:2], sep='.')")
```

File: README.md [hunk 2]
```diff
@@ -221,20 +221,21 @@ but [U+2502/"Box Drawings Light Vertical"](https://www.compart.com/en/unicode/U+
...
 ```
 
-Or this line in your `~/.config/fish/config.fish` file:
+Or these lines to your `~/.config/fish/config.fish` file:
 
 ```sh
-set -gx PY_PYTHON (head -n 1 (pyenv root)/version | cut -d "." -f 1,2)
+set -gx PY_PYTHON (pyenv exec python -c "import sys; print('.'.join(map(str, sys.version_info[0:2])))")
+set -gx PY_PYTHON3 (pyenv exec python3 -c "import sys; print(*sys.version_info[0:2], sep='.')")
```
Comment (brettcannon) [inline README.md:238]:
```suggestion
set -gx PY_PYTHON (pyenv exec python3 -c "import sys; print(*sys.version_info[0:2], sep='.')")
```

**Review by brettcannon — CHANGES_REQUESTED** [review]
One tricky thing with this compared to the old solution is `PY_PYTHON` is the default version to use when no specific major version is requested, e.g. it's not the same as `PY_PYTHON2`. That means this isn't technically right as you have to make an assumption as to what major version of Python people want to depend on compared to the solution of reading `(pyenv root)/version`).

Maybe it's better to list both solutions?

**Comment by cas--** [comment]
I merged your suggestions but then changed my mind and reverted as I think there has been a misunderstanding.

> One tricky thing with this compared to the old solution is PY_PYTHON is the default version to use when no specific major version is requested e.g. it's not the same as PY_PYTHON2

The use of `python` is not meant to refer to `python2`. Using `$(pyenv exec python)` solution mirrors the old solution since pyenv `python` is what the user set via pyenv global (the `(pyenv root)/version` file) or fallback to system. 

I hope this make sense?

**Comment by brettcannon** [comment]
Then please drop the `PY_PYTHON3` example as I don't think many people will care enough to warrant making the example more complex.

File: README.md [hunk 3]
```diff
@@ -221,20 +221,21 @@ but [U+2502/"Box Drawings Light Vertical"](https://www.compart.com/en/unicode/U+
...
-[pyenv version file](https://github.com/pyenv/pyenv#choosing-the-python-version).
+versions, you'll want to set the version the Launcher to the pyenv 
+[global version](https://github.com/pyenv/pyenv/blob/master/COMMANDS.md#pyenv-global).
 
-You can add this line to your `.zshrc` or `.bashrc` file:
+Add these lines to your `.zshrc` or `.bashrc` file:
 
 ```sh
-export PY_PYTHON=$(head -n 1 $(pyenv root)/version | cut -d "." -f 1,2)
+export PY_PYTHON=$(pyenv exec python -c "import sys; print('.'.join(map(str, sys.version_info[0:2])))")
```
Comment (brettcannon) [inline README.md:230]:
```suggestion
export PY_PYTHON=$(pyenv exec python -c "import sys; print('.'.join(map(str, sys.version_info[:2])))")
```

File: README.md (hunk 1 above)
Comment (brettcannon) [inline README.md:231]:
```suggestion
```

File: README.md (hunk 2 above)
Comment (brettcannon) [inline README.md:238]:
```suggestion
```

File: README.md [hunk 4]
```diff
@@ -221,20 +221,21 @@ but [U+2502/"Box Drawings Light Vertical"](https://www.compart.com/en/unicode/U+
...
+export PY_PYTHON3=$(pyenv exec python3 -c "import sys; print(*sys.version_info[0:2], sep='.')")
 ```
 
-Or this line in your `~/.config/fish/config.fish` file:
+Or these lines to your `~/.config/fish/config.fish` file:
 
 ```sh
-set -gx PY_PYTHON (head -n 1 (pyenv root)/version | cut -d "." -f 1,2)
+set -gx PY_PYTHON (pyenv exec python -c "import sys; print('.'.join(map(str, sys.version_info[0:2])))")
```
Comment (brettcannon) [inline README.md:237]:
```suggestion
set -gx PY_PYTHON (pyenv exec python -c "import sys; print('.'.join(map(str, sys.version_info[:2])))")
```

**Review by brettcannon — CHANGES_REQUESTED** [review]

**Comment by cas--** [comment]
I agree, was including it for completeness for `-3` option

File: README.md [hunk 5]
```diff
@@ -221,20 +221,19 @@ but [U+2502/"Box Drawings Light Vertical"](https://www.compart.com/en/unicode/U+
 ### How can I make the Python Launcher use my default Python version from pyenv?
 
 If you're using [pyenv](https://github.com/pyenv/pyenv) to manage your Python
-versions, you'll want to grab the major and minor version from the first Python
-version listed in your default
-[pyenv version file](https://github.com/pyenv/pyenv#choosing-the-python-version).
+versions, you'll want to set the version the Launcher to the pyenv
```
Comment (brettcannon) [inline README.md:224]:
```suggestion
versions, you'll want to set the version the Launcher uses to the pyenv 
```

File: README.md [hunk 6]
```diff
@@ -221,20 +221,19 @@ but [U+2502/"Box Drawings Light Vertical"](https://www.compart.com/en/unicode/U+
...
 
 If you're using [pyenv](https://github.com/pyenv/pyenv) to manage your Python
-versions, you'll want to grab the major and minor version from the first Python
-version listed in your default
-[pyenv version file](https://github.com/pyenv/pyenv#choosing-the-python-version).
+versions, you'll want to set the version the Launcher to the pyenv 
+[global version](https://github.com/pyenv/pyenv/blob/master/COMMANDS.md#pyenv-global).
 
-You can add this line to your `.zshrc` or `.bashrc` file:
+Add these lines to your `.zshrc` or `.bashrc` file:
```
Comment (brettcannon) [inline README.md:227]:
```suggestion
Add this line to your `.zshrc` or `.bashrc` file:
```

File: README.md [hunk 7]
```diff
@@ -221,20 +221,19 @@ but [U+2502/"Box Drawings Light Vertical"](https://www.compart.com/en/unicode/U+
...
 
 ```sh
-export PY_PYTHON=$(head -n 1 $(pyenv root)/version | cut -d "." -f 1,2)
+export PY_PYTHON=$(pyenv exec python -c "import sys; print('.'.join(map(str, sys.version_info[:2])))")
 ```
 
-Or this line in your `~/.config/fish/config.fish` file:
+Or these lines to your `~/.config/fish/config.fish` file:
```
Comment (brettcannon) [inline README.md:233]:
```suggestion
Or this line to your `~/.config/fish/config.fish` file:
```

**Review by brettcannon — CHANGES_REQUESTED** [review]

**Review by brettcannon — APPROVED** [review]

**Review by brettcannon — CHANGES_REQUESTED** [review]
I just realized the changes were made against the main copy of the README and not the template which the README is generated from. Would you mind applying the changes to https://github.com/brettcannon/python-launcher/blob/main/docs/readme/template.md instead?

**Comment by brettcannon** [comment]
Thanks for the help with this!

**Pull request merged**